src/hydro/hydro_driver.cpp

```diff
@@ -150,6 +226,47 @@ TaskCollection HydroDriver::MakeTaskCollection(BlockList_t &blocks, int stage) {
         hydro_pkg.get());
   }
 
+  // Calculate 1D profile of cooling rate
+  if (stage == 1) {
+    auto pkg = blocks[0]->packages.Get("Hydro");
+
+    AllReduce<PinnedArray1D<Real>> *pview_reduce =
+        pkg->MutableParam<AllReduce<PinnedArray1D<Real>>>("profile_reduce");
+
+    // initialize values to zero
+    for (int i = 0; i < pview_reduce->val.size(); i++) {
+      pview_reduce->val(i) = 0;
+    }
+
+    // create task region
+    int reg_dep_id;
+    TaskRegion &solver_region = tc.AddRegion(num_partitions);
+
+    for (int i = 0; i < num_partitions; i++) {
+      reg_dep_id = 0;
+      TaskList &tl = solver_region[i];
+
+      // compute rank-local reduction
+      auto &mu0 = pmesh->mesh_data.GetOrAdd("base", i);
+      TaskID local_sum =
+          (i == 0 ? tl.AddTask(none, CalculateCoolingRateProfile, mu0.get()) : none);
```

Comment:
The "local" reduction should happen for all partitions, don't they?
Also this task should probably be the region dependency, so that all list in this regions are done with this tasks before the global reduction can start.

Reply:
Yes, I agree I both that this should be done for all partitions, since I am getting the data only for partition $i$. I meant to get the data for all MeshBlocks, but I didn't actually do that. Also set the region dependency on the wrong task. I've followed your example code and I think I've fixed it.

Reply:
Well, this does not fix the observed issue.